### PR TITLE
Set missing component properties to default value

### PR DESCRIPTION
**Pull request by Ersikan**

### CODE REVIEW

File: libecs/tests/Core/Component/test_tag.cxx [hunk 1]
```diff
@@ -26,5 +26,5 @@ TEST_CASE("Correct description for tag component", "[component][tag]")
     REQUIRE(value == Value{Record{{"name", new_tag_name}}});
 
     REQUIRE_THROWS(Helpers<Tag>::updateTypeWithValue(tag, Value{3}));
-    REQUIRE_THROWS(Helpers<Tag>::updateTypeWithValue(tag, Value{Record{}}));
+    // REQUIRE_THROWS(Helpers<Tag>::updateTypeWithValue(tag, Value{Record{}}));
```
Comment (zcorniere) [inline libecs/tests/Core/Component/test_tag.cxx:29]: Commented test

File: libecs/source/Core/Data/type.cxx [hunk 2]
```diff
@@ -64,4 +64,18 @@ data::Value Type::defaultValue() const
         },
         static_cast<const Type::variant &>(*this));
 }
+
+bool Type::isSubsetOf(const Type &other) const
+{
+    if (const RecordType *record = std::get_if<RecordType>(this)) {
```
Comment (zcorniere) [inline libecs/source/Core/Data/type.cxx:70]: if *this == other, does that mean this test will fail ?
If it doesn't, perhaps move the test at line 79 up in order to exit early.

File: libecs/include/pivot/ecs/Core/Component/description_helpers_impl.hxx [hunk 3]
```diff
@@ -65,15 +65,18 @@ struct Helpers<T> {
 
     static void updateTypeWithValue(T &data, const data::Value &value)
     {
+        data = T{};
```
Comment (zcorniere) [inline libecs/include/pivot/ecs/Core/Component/description_helpers_impl.hxx:68]: Is it necessary to reset the whole object ? Can be expensive. Perhaps reset individual field if the value is not found

File: libecs/source/Core/Data/type.cxx [hunk 4]
```diff
@@ -64,4 +64,18 @@ data::Value Type::defaultValue() const
...
+        if (const RecordType *other_record = std::get_if<RecordType>(&other)) {
+            for (auto &[key, subtype]: *record) {
+                if (!other_record->contains(key)) { return false; }
+                if (other_record->at(key) != subtype) { return false; }
```
Comment (zcorniere) [inline libecs/source/Core/Data/type.cxx:74]: Instead of a contains/at combo, use find

File: libecs/source/Core/Data/type.cxx (hunk 2 above)
Comment (Ersikan) [reply libecs/source/Core/Data/type.cxx:70]: The function can return true even when *this == other is false !

File: libecs/include/pivot/ecs/Core/Component/description_helpers_impl.hxx (hunk 3 above)
Comment (Ersikan) [reply libecs/include/pivot/ecs/Core/Component/description_helpers_impl.hxx:68]: How do you get the default value of only one property of a class/struct then ? For example for Transform I want to default value:
```c++
class Transform
{
public:
    glm::vec3 position = glm::vec3(0.0f);
    glm::vec3 rotation = glm::vec3(0.0f);
    glm::vec3 scale = glm::vec3(1.0f);
};
```

**Review by zcorniere — APPROVED** [review]

**Pull request merged**